overlord/configstate/helpers.go

```diff
@@ -0,0 +1,40 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configstate
+
+import (
+	"sort"
+	"strings"
+)
+
+func sortPatchKeys(patch map[string]interface{}) []string {
+	depths := make(map[string]int, len(patch))
+	var keys []string
+	for k := range patch {
+		parts := strings.Split(k, ".")
```

Comment:
I think using `strings.Count` might be a bit cheaper here.

Reply:
Indeed, I thought `Split` would make the intent more clear, and the amounts of data we're dealing with here is negligible; I can change this if you think it's better to have Count?

Reply:
Changed, per similiar suggestion from Maciej.